src/Pool.sol

```diff
@@ -538,18 +516,40 @@ abstract contract Pool {
                 newIdx++;
             }
             _setDelta(receiver, _currTimestamp(), newAmt - currAmt);
-            updates.updates[updates.length++] = StreamUpdate({
-                receiver: receiver,
-                amtPerSec: uint128(newAmt),
-                endTime: newAmt == 0 ? _currTimestamp() : newEndTime
-            });
+            emitStreamUpdated(senderId, receiver, uint128(newAmt), newEndTime);
             // The receiver was never used, initialize it.
             if (!pickCurr && receiverStates[receiver].nextCollectedCycle == 0) {
                 receiverStates[receiver].nextCollectedCycle = _currTimestamp() / cycleSecs + 1;
             }
         }
     }
 
+    /// @notice Emit a relevant event when a stream is updated.
+    /// @param senderId The id of the sender of the updated stream
+    /// @param receiver The receiver of the updated stream
+    /// @param amtPerSec The new amount per second sent from the sender to the receiver
+    /// or 0 if sending is stopped
+    /// @param endTime The timestamp when the funds stop being sent.
+    function emitStreamUpdated(
+        SenderId memory senderId,
+        address receiver,
+        uint128 amtPerSec,
+        uint64 endTime
+    ) internal {
+        if (amtPerSec == 0) endTime = _currTimestamp();
```

Comment:
I would move this line to the `_updateStreams`. For me the emit function should just dealing with emitting the correct event.

Reply:
IMO it's the best place for this logic. It simplifies the API of `emitStreamUpdated` and guarantees consistency, because now it's impossible to forget about this special case. It's also logic specific for emitting the event, not for updating streams, so it's a good separation of concerns.